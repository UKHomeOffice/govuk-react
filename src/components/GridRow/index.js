// INSERT A COMMENT REFERENCE TO EXTERNAL URL IF POSSIBLE

import React from "react";
import PropTypes from "prop-types";
import glamorous from "glamorous";
import { MEDIA_QUERIES, SITE_WIDTH } from "../../constants/index";

const GridRowInner = glamorous.div({
  margin: "0 -15px",
  width: "100%",
  [MEDIA_QUERIES.LARGESCREEN]: {
    display: "flex",
    maxWidth: SITE_WIDTH
  }
});

const GridRow = ({ children }) => <GridRowInner>{children}</GridRowInner>;

GridRow.propTypes = {
  children: PropTypes.node.isRequired
};

export default GridRow;
