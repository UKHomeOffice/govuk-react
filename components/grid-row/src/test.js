import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';

import GridRow from './';

expect.addSnapshotSerializer(createSerializer(emotion));

describe(GridRow, () => {
  const wrapper = <GridRow />;

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<GridRow>Example</GridRow>, div);
  });

  it('matches snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('enzyme.mount');
  });
});
