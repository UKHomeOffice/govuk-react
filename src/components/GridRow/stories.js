import React from "react";
import { storiesOf } from "@storybook/react";

import GridRow from ".";

storiesOf("GridRow", module).add("GridRow", () => (
  <GridRow>GridRow example</GridRow>
));
