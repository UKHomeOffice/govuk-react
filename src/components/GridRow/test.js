import React from "react";
import ReactDOM from "react-dom";
import GridRow from "./";

describe(GridRow, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<GridRow>Example</GridRow>, div);
  });
});
