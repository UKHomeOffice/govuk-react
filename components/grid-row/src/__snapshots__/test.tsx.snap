// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridRow matches snapshot: enzyme.mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 -15px;
}

@media only screen and (min-width:641px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<GridRow>
  <src__StyledRow>
    <div
      className="c0"
    >
      Example
    </div>
  </src__StyledRow>
</GridRow>
`;
